accepted/2021/package-validation/package-validation.md

```diff
@@ -0,0 +1,236 @@
+breaking change. They solve this problem by adding an overload instead:
+
+```C#
+public static Client Connect(string url)
+{
+    return Client(url, Timeout.InfiniteTimeSpan);
+}
+
+public static Client Connect(string url, TimeSpan timeout)
+{
+    // ...
+}
+```
+
+### Making an intentional breaking change
+
+Armani is working on version 0.7 of their library. While they don't want to make
+accidental breaking changes, they do want to make changes based on user feedback.
+
+In version 0.6 they had a `Connect()` method that accepted a timeout as an
+`int`. Users complaint that timeouts in .NET are nowadays generally modelled
+using `TimeSpan` instead. After making the necessary code change, rebuilding the
+library results in an error:
+
+> error SYSLIB1234: The method 'Connect(string, int)' exists in the previous
+> version of the NuGet package (Fabrikam.Client 0.6.3) but no longer exists in
+> the current version being built (0.7.0). This is a breaking change.
+
+Since Armani's library is still an 0.x, which, according to SemVer, doesn't
+promise a stable API, they briefly consider turning off the breaking change
+validation entirely. However, Armani feels that even though it's 0.x they still
+don't want to make accidental breaking change so they decide to just suppress
+this specific instance of the breaking change:
+
+```C#
+[SuppressMessage("Interoperability", "SYSLIB1234")]]
+public static Client Connect(string url, TimeSpan timeout)
+{
+    // ...
+}
+```
+
+## Requirements
+
+### Goals
+
+* Performance should be good so that basic package validation can be on by
+  default (when the project generates a package), but it must be possible to
+  turn it off as well
+* Comparing to previous version should be opt-in because it requires downloading
+  additional files
+* Developers need to be able to make intentional breaking changes while still
+  validating that they didn't make other unintentional breaking changes.
+
+### Non-Goals
+
+* Having zero impact on performance (as that's not feasible)
+
+## Stakeholders and Reviewers
+
+* Libraries team
+* NuGet team
+* SDK team
+
+## Design
+
+### API Compat Tooling
+
+The API compat tooling is described in a separate [spec][api-compat-spec].
+
+### Diagnostic IDs
+
+***OPEN QUESTION**: Should we introduce a new prefix, such as `APC` (ApiCompat)
+or `PKG` (packaging) or should we reuse `CA`/`SYSLIB`?*
+
+### Previous version configuration
+
+We need to be able to configure the following things:
+
+1. Should a previous version be validated?
+2. Which feeds should be scanned? Unless the user configures a different feed,
+   we'd use nuget.org.
+
+I don't believe we need to know the previous version. The previous version would
+be defined on a per-feed basis. In any feed, the previous version is the one
+that immediately precedes the version being built.
+
+***OPEN QUESTION**: Do we need a SemVer setting? Originally I thought about a
```

Comment:
After reading the proposal, this tool sounds like it is fixing a case where authors do not respect SemVer & should be warned that they are doing so.

Perhaps the tool should be somewhat "SemVer aware" so that it can assess the potential disruption the author can cause by not incrementing the respective version number.

"You made an incompatible API change detected by our tool, you should bump your major version."

"You made a backwards compatible API change detected by our tool, you should bump your minor version."

This might help empower users to embrace SemVer further with actionable steps to take or ignore.